receiver/splunkhecreceiver/receiver.go

```diff
@@ -231,20 +253,75 @@ func (r *splunkReceiver) Shutdown(context.Context) error {
 	return err
 }
 
-func (r *splunkReceiver) writeSuccessResponse(ctx context.Context, resp http.ResponseWriter, eventCount int) {
+func (r *splunkReceiver) writeSuccessResponseWithAck(ctx context.Context, resp http.ResponseWriter, eventCount int, partitionID string) {
+	ackID := r.ackExt.ProcessEvent(partitionID)
```

Comment:
I know your usages are safe thus far, but it may be wise to add another nil check on the ack extension here to ensure it's set as expected.